src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -57,18 +57,18 @@ bool ignoreCache
 		if( authnResponse is AuthenticateResponse.MfaRequired mfaInfo ) {
 			OktaMfaFactor mfaFactor = consolePrompter.SelectMfa( mfaInfo.Factors );
 
-			if( !IsMfaFactorTypeSupported( mfaFactor.FactorType ) ) {
+			if( mfaFactor.FactorName == "Unknown" ) {
```

Comment:
Not a fan of this at all

Reply:
This can be a type check against the (maybe renamed) base type, or at least you can make the string a constant so there's no possibility of a typo.

Reply:
Doesn't look great IMO. https://github.com/Brightspace/bmx/pull/459/commits/19413d471f72c72cba5822489427766baa96cec9

I think we should instead rely on something using ``UnknownDerivedTypeHandling = JsonUnknownDerivedTypeHandling.FailSerialization``
& filtering out failures seems better although definitely more involved code wise (de-serialization call for this API response handling will need rework).

After all we aren't proceeding when unknown factor type is selected, might as well not populate the list with them (maybe a separate message for "x, y factors are not supported by BMX"

Reply:
I was thinking of giving "unknown" a special constant, rather than reusing the "FactoryType" thingy we're using elsewhere. Reusing that would be confusing for sure.

What about doing a runtime type check though? We could call the base type something like `OktaMfaFactorUnspecified`, and check the exact type (rather than using `is` which considers child types a match).

More involved code change is precisely why I didn't suggest failing serialization directly, but that's an option for sure. We can even make the base type abstract.

Reply:
actually, with a better base type name, the constant check wouldn't look so confusing either.

Reply:
Think we should go with a standalone const + factor name check, 

https://github.com/Brightspace/bmx/pull/459/commits/b57bf153c866c1ed23d077d374a5cd4448073e1e is confusing in it's own right (since we use the base type name in quite a few contexts, it is awkward to name it "unknown"/"unsupported".

Reply:
Maybe ```BaseOktaMfaFactor```  or something might help, but it still feels wrong.

Reply:
Yeah, I see. Renaming the base type doesn't work. A distinctly named constant is good. The `FactorType` constant isn't in the type hierarchy anyway.

Reply:
Changed to a better named const for this, should revisit if this ends up happening https://github.com/dotnet/runtime/issues/93471